packages/components/src/calendar/test/index.tsx

```diff
@@ -0,0 +1,751 @@
+				expect(
+					getDateButton( new Date( today.getFullYear(), today.getMonth(), date ) )
+				).toBeDisabled();
+			}
+			expect( getDateButton( yesterday ) ).not.toBeDisabled();
+			expect( getDateButton( today ) ).not.toBeDisabled();
+			expect( getDateButton( addDays( today, 1 ) ) ).not.toBeDisabled();
+
+			for ( date = today.getDate() + 2; date < 32; date++ ) {
+				expect(
+					getDateButton( new Date( today.getFullYear(), today.getMonth(), date ) )
+				).toBeDisabled();
+			}
+		} );
+
+		it( 'should support disabling all dates within a certain date range via the `disabled` prop', async () => {
+			render( <DateCalendar disabled={ { from: yesterday, to: addDays( today, 1 ) } } /> );
+
+			let date;
+
+			for ( date = 1; date < today.getDate() - 1; date++ ) {
+				expect(
+					getDateButton( new Date( today.getFullYear(), today.getMonth(), date ) )
+				).not.toBeDisabled();
+			}
+			expect( getDateButton( yesterday ) ).toBeDisabled();
+			expect( getDateButton( today ) ).toBeDisabled();
+			expect( getDateButton( addDays( today, 1 ) ) ).toBeDisabled();
+
+			for ( date = today.getDate() + 2; date < 32; date++ ) {
+				expect(
+					getDateButton( new Date( today.getFullYear(), today.getMonth(), date ) )
+				).not.toBeDisabled();
+			}
+		} );
+
+		it( 'should support disabling specific days of the week via the `disabled` prop', async () => {
+			const weekendsInMay = [ 3, 4, 10, 11, 17, 18, 24, 25, 31 ];
+			render( <DateCalendar disabled={ { dayOfWeek: [ 0, 6 ] } } /> );
+
+			for ( const date of weekendsInMay ) {
+				expect(
+					getDateButton( new Date( today.getFullYear(), today.getMonth(), date ) )
+				).toBeDisabled();
+			}
+		} );
+
+		it( 'should disable the previous and next months buttons if the `disableNavigation` is set to `true`', async () => {
+			const user = setupUserEvent();
+
+			render( <DateCalendar disableNavigation /> );
+
+			expect( screen.getByRole( 'button', { name: /previous month/i } ) ).toHaveAttribute(
+				'aria-disabled',
+				'true'
+			);
+			expect( screen.getByRole( 'button', { name: /next month/i } ) ).toHaveAttribute(
+				'aria-disabled',
+				'true'
+			);
+
+			await user.tab();
+			expect( screen.getByRole( 'button', { name: /today/i } ) ).toHaveFocus();
+		} );
+	} );
+
+	// Note: we're not testing localization of strings. We're only testing
+	// that the date formatting, computed dir, and calendar format are correct.
+	describe( 'Localization', () => {
+		it( 'should localize the calendar based on the `locale` prop', async () => {
+			const user = setupUserEvent();
+
+			render( <DateCalendar locale={ ar } /> );
+
+			// Check computed writing direction
+			expect( screen.getByRole( 'application', { name: 'Date calendar' } ) ).toHaveAttribute(
+				'dir',
+				'rtl'
+			);
+
+			// Check month name
+			const grid = screen.getByRole( 'grid', {
+				name: monthNameFormatter( 'ar' ).format( today ),
+			} );
+			expect( grid ).toBeInTheDocument();
```

Comment:
User should be seeing this, so this might be a better assertion to use:

```suggestion
			expect( grid ).toBeVisible();
```